botocore/hooks.py

```diff
@@ -378,8 +416,12 @@ def unregister(self, event_name, *args, **kwargs):
         )
 
     def _register(self, func, event_name, *args, **kwargs):
-        for endpoint_prefix, service_ids in DUPLICATE_EVENTS.items():
-            event_parts = event_name.split('.')
+        event_parts = event_name.split('.')
+        if not event_parts[0] in self._duplicate_event_whitelist:
+            self._alias(func, event_name, *args, **kwargs)
+            return
+
+        for endpoint_prefix, service_ids in self._duplicate_events.items():
```

Comment:
We may want to rename this not to be specific to `endpoint_prefix` and `service_id` as the `duplicate_events` may not necessarily be related to endpoint prefixes and service ID's.